portal/v2/src/ClusterDetailListComponents/Statistics/GraphOptionsComponent.tsx

```diff
@@ -0,0 +1,220 @@
+      var localDate = new Date()
+      localDate.setUTCFullYear(props.endDate.getFullYear(), props.endDate.getMonth(), props.endDate.getDate())
+      localDate.setUTCHours(date.getHours());
+      localDate.setUTCMinutes(date.getMinutes());
+      props.setEndDate(localDate)        
+     };
+    const onDateChange = (date: Date | null | undefined): void => { 
+      let localDate: Date = new Date();
+      localDate.setUTCFullYear(date!.getFullYear(), date!.getMonth(), date!.getDate())
+      localDate.setHours(props.endDate.getHours());
+      localDate.setMinutes(props.endDate.getMinutes());
+      props.setEndDate(localDate)  
+     };
+     
+    return (
+      <>
+      <Stack horizontal verticalAlign="center">
+      <Stack horizontal verticalAlign="center" className={classNames.iconContainer} /* style={{ boxShadow: theme.effects.elevation8 }}*/>
+          <TooltipHost>
+            <IconButton
+              onClick={_decreaseDuration}
+              iconProps={calculatorSubtract}
+              styles={iconButtonStyles}
+            />
+          </TooltipHost>
+          <TooltipHost>
+            <Text styles={durationStyle}>{props.duration}</Text>
+          </TooltipHost>
+          <TooltipHost >
+            <IconButton
+              onClick={_increaseDuration}
+              iconProps={calculatorAddition}
+              styles={iconButtonStyles}
+            />
+          </TooltipHost>
+        </Stack>
+        <Stack horizontal verticalAlign="center" className={dateTimePickerStyles.iconContainer}  /* style={{ boxShadow: theme.effects.elevation8 }}*/>
+          <DatePicker
+            styles={datePickerStyles}
+            placeholder="End Date"
+            ariaLabel="End Date"
+            onSelectDate={onDateChange}
+            value={convertToUTC(props.endDate)}
+            allowTextInput
+          />
+          <TimePicker
+            styles={timePickerStyles}
+            allowFreeform
+            placeholder={timeToString(convertToUTC(props.endDate))}
+            autoComplete="on"
+            onChange={onTimeChange}
+            defaultValue={convertToUTC(props.endDate)}
+            useComboBoxAsMenuWidth
+          />
+        </Stack>
+      </Stack>
+      </>
+    )
+  }
+
+  function timeToString(date: Date): string {
+    var str
+    let hourString = date.getHours().toString()
+    if (hourString.length === 1) {
+      str = "0" + hourString + ":"
+    } else {
+      str = hourString + ":"
+    }
+    let minuteString = date.getMinutes().toString()
+    if (minuteString.length === 1) {
+      str = str + "0" + minuteString
+    } else {
+      str += minuteString
+    }
+
+    return str
+  }
+
+  export const convertToUTC = (date: Date): Date => {
+    let localDate = new Date()
+    localDate.setFullYear(date.getUTCFullYear(), date.getUTCMonth(), date.getUTCDate())
+    localDate.setHours(date.getUTCHours())
+    localDate.setMinutes(date.getUTCMinutes())
+    
+    return localDate
+  }
```

Comment:
[GraphOptionsComponent.tsx](https://github.com/Azure/ARO-RP/pull/2674/files#diff-98976a43957c74e6d37bd36960c9565126b2f9abbbeadae63f19212660d27892) Contains similar or identical code. Consider turning it into a util function and importing it in both places.

Reply:
In that code the time/date is actually getting changed, here the time is only getting converted to UTC without any change.